add rhcos section (#15)
elmiko/ocp-infrastructure-provider-onboarding-guide: docs/rhcos/index.md

# RHCOS Platform Support

> To support a new platform one of the main requirements is to have RedHat CoreOS(RHCOS) operating system work on the platform for provisioning the machines. This [getting started guide](https://docs.fedoraproject.org/en-US/fedora-coreos/getting-started/) gives a good overview of the Fedora CoreOS(FCOS) operating system and way to provision machines on different platforms. The process of building RHCOS and FCOS is largely the same except for the packages contained inside of the OS itself (RHEL vs Fedora).

Reviewer: minor nit
```suggestion
To support a new platform one of the main requirements is to have Red Hat CoreOS(RHCOS) operating system work on the platform for provisioning the machines. This [getting started guide](https://docs.fedoraproject.org/en-US/fedora-coreos/getting-started/) gives a good overview of the Fedora CoreOS(FCOS) operating system and way to provision machines on different platforms. The process of building RHCOS and FCOS is largely the same except for the packages contained inside of the OS itself (RHEL vs Fedora).
```

> To support a new platform one of the main requirements is to have Red Hat CoreOS(RHCOS) operating system work on the platform for provisioning the machines. This [getting started guide](https://docs.fedoraproject.org/en-US/fedora-coreos/getting-started/) gives a good overview of the Fedora CoreOS(FCOS) operating system and way to provision machines on different platforms. The process of building RHCOS and FCOS is largely the same except for the packages contained inside of the OS itself (RHEL vs Fedora).

Reviewer 2: ```suggestion
To support a new platform one of the main requirements is to have Red Hat Enterprise Linux CoreOS (RHCOS) operating system work on the platform for provisioning the machines. This [getting started guide](https://docs.fedoraproject.org/en-US/fedora-coreos/getting-started/) gives a good overview of the Fedora CoreOS (FCOS) operating system and way to provision machines on different platforms. The process of building RHCOS and FCOS is largely the same except for the packages contained inside of the OS itself (RHEL vs Fedora).
```

## File a ticket for new platform support

Please make sure to file a ticket against [Fedora-CoreOS-Tracker](https://github.com/coreos/fedora-coreos-tracker.git) by filing a new issue and choosing the [Request a new platform](https://github.com/coreos/fedora-coreos-tracker/issues/new/choose) option.
Some important questions to be thought about/answered in the platform request tracker are:

> - The official name of the platform - this is name which will identify your platform across all Openshift repos and APIs

Reviewer 2: I could be wrong, but I think the CoreOS platform name doesn't always match the OpenShift one.

Author: hmm..yeah that's correct. the platform id in the kargs is not necessarily the platform used in the Openshift APIs...will call that out. thanks!

> - How is userdata provided - this refers to how [ignition](https://github.com/coreos/ignition) configuration is provided for this platform.

Reviewer 2: ```suggestion
- How is userdata provided - this refers to how [Ignition](https://github.com/coreos/ignition) configuration is provided for this platform.
```

- [Working with CoreOS Assembler guide](https://github.com/coreos/coreos-assembler/blob/main/docs/working.md)
- [Building Fedora CoreOS](https://github.com/coreos/coreos-assembler/blob/main/docs/building-fcos.md)
- [Working on CoreOS Assembler guide](https://github.com/coreos/coreos-assembler/blob/main/docs/devel.md)
- [CoreOS Assembler Design](https://github.com/coreos/coreos-assembler/blob/main/docs/design.md)

In addition, there are tools that help you test your image locally and also help you upload your images to the cloud. [Mantle](https://github.com/coreos/coreos-assembler/tree/main/mantle#mantle-gluing-container-linux-together) inside the coreos-assembler has a bunch of tools which help with different things:

- kola - for launching instances and running tests
- kolet - an agent for kola that runs on instances
- ore - for interfacing with cloud providers

## Ignition support

> [ignition](https://github.com/coreos/ignition#ignition) is a utility that configures the machine. A list of supported platforms and how they provide the configuration to ignition is listed [here](https://github.com/coreos/ignition/blob/main/docs/supported-platforms.md). Whenever a new platform is added, corresponding code must be added to specify the way to [fetch configuration](https://github.com/coreos/ignition/tree/main/internal/providers) for the platform.

Reviewer 2: ```suggestion
[Ignition](https://github.com/coreos/ignition#ignition) is a utility that configures the machine. A list of supported platforms and how they provide the configuration to Ignition is listed [here](https://github.com/coreos/ignition/blob/main/docs/supported-platforms.md). Whenever a new platform is added, corresponding code must be added to specify the way to [fetch configuration](https://github.com/coreos/ignition/tree/main/internal/providers) for the platform.
```

## Afterburn support

> [afterburn](https://github.com/coreos/afterburn) is used to implement cloud provider specific functionality by interacting with its metadata endpoints. It can be used to optionally set things like hostname and also inject network kernel command line arguments. In addition, it is also used to retrieve attributes from the metadata. Once added, some of this functionality can be enabled as systemd units in the machine config operator.

Reviewer 2: ```suggestion
[Afterburn](https://github.com/coreos/afterburn) is used to implement cloud provider specific functionality by interacting with its metadata endpoints. It can be used to optionally set things like hostname and also inject network kernel command line arguments. In addition, it is also used to retrieve attributes from the metadata. Once added, some of this functionality can be enabled as systemd units in the Machine Config Operator.
```

## Questions

> Questions can be directed to the OpenShift Slack channel **#forum-coreos**

Reviewer: which slack server is this on? (remember these docs will be public facing)

Author: ah..yes..the slack channel is CoreOS slack workspace...let me mention that here..is that what you are asking ?

Reviewer: well, we could mention this channel but make sure to be explicit about it being for red hat associates. i'm not sure if that adds value here though.

Author: i have linked to the Fedora CoreOS communication channels as Colin mentioned.

> Questions can be directed to the OpenShift Slack channel in the CoreOS workspace(coreos.slack.com) **#forum-coreos**

Reviewer 3: I'd note this is internal to RHT, and it's also often better if possible to work in FCOS first in public.  https://github.com/coreos/fedora-coreos-tracker#communication-channels-for-fedora-coreos has the communication channel links.

Author: makes sense! updated. thanks!
